src/geocel/rasterize/Raytracer.hh

```diff
@@ -0,0 +1,182 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file geocel/rasterize/Raytracer.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/math/Algorithms.hh"
+
+#include "ImageLineView.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Trace each pixel along a line.
+ *
+ * \tparam GTV Geometry Track View
+ * \tparam F   Calculate the pixel value (result_type) given GTV
+ */
+template<class GTV, class F>
+class Raytracer
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using result_type = int;
+    //!@}
+
+  public:
+    // Construct from geo track view, function to calculate ID, and image
+    inline CELER_FUNCTION
+    Raytracer(GTV&& geo, F&& calc_id, ImageLineView const& image);
+
+    // Calculate the value for the next pixel
+    inline CELER_FUNCTION result_type operator()(size_type pix);
+
+  private:
+    //// DATA ////
+
+    GTV geo_;
+    F calc_id_;
+    ImageLineView const& image_;
+
+    size_type pixel_;
+
+    //// HELPER FUNCTIONS ////
+
+    // Initialize the geometry at this pixel index
+    inline CELER_FUNCTION void initialize_at_pixel(size_type pix);
+
+    //! Sentinel value for needing the pixel to be reset
+    static CELER_CONSTEXPR_FUNCTION size_type invalid_pixel()
+    {
+        return static_cast<size_type>(-1);
+    }
+
+    //! Stop tracking the pixel if this many crossings occur
+    static CELER_CONSTEXPR_FUNCTION short int max_crossings_per_pixel()
+    {
+        return 32;
```

Comment:
Should this be a constexpr somewhere more official, rather than a "magic number"? Also, I could see this being something that needs to be configurable for a suitably zoom-out ray trace of a geometry with fine features.

Reply:
I agree it should be configurable; previously it was a hard-coded constant in the implementation code itself. This way at least documents the capability at the class level; later we can extend the implementation to make it configurable.